RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var retrieved = await _context.Albums.FindAsync(id);
+
+            var relatedAlbums = _context.Albums
```

Comment:
I really like the null check of id in the Show action. I would recommend implementing another null check for the variable relatedAlbums so ensure it is not null before it is used to assign the viewModel variable that gets passed to the View.